packages/eslint-plugin-react-hooks/src/ExhaustiveDeps.js

```diff
@@ -218,8 +224,13 @@ export default {
         }
         const id = def.node.id;
         const {name} = callee;
-        if (name === 'useRef' && id.type === 'Identifier') {
-          // useRef() return value is stable.
+        // Check if the hook is named `useRef` or one of the other hooks that's
+        // been deemed "safe"
+        if (
```

Comment:
Suggestion: Moving the logic into a named function would remove the need for the comment and increase readability?

```
function isHookSafe(id, name, safeHooks) {
  const isSafeHook = safeHooks && safeHooks.test(name);
  const isUseRef = name === 'useRef';

  return id.type === 'Identifier' && (isSafeHook || isUseRef);
}

```

Then the if statement can be: 
```
  if (isHookSafe(id, name, safeHooks) {
    return true;
  } else if (name === 'useState' || name === 'useReducer') {
  ...
```

Reply:
Done. I rearranged the body a bit though to keep the `id.type === ...` check first, so that we're not running the regex against every AST node that we hit.